cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPlatformResources.java

```diff
@@ -657,23 +660,14 @@ public CloudAccessConfigs accessConfigs(CloudCredential cloudCredential, Region
         AwsCredentialView awsCredentialView = new AwsCredentialView(cloudCredential);
         AmazonIdentityManagement client = awsClient.createAmazonIdentityManagement(awsCredentialView);
         try {
-            ListInstanceProfilesRequest listInstanceProfilesRequest = new ListInstanceProfilesRequest();
-            listInstanceProfilesRequest.setMaxItems(MAX_ITEMS);
-            ListInstanceProfilesResult listRolesResult = client.listInstanceProfiles(listInstanceProfilesRequest);
-            for (InstanceProfile instanceProfile : listRolesResult.getInstanceProfiles()) {
-                Map<String, Object> properties = new HashMap<>();
-                properties.put("arn", instanceProfile.getArn());
-                properties.put("creationDate", instanceProfile.getCreateDate().toString());
-                if (!instanceProfile.getRoles().isEmpty()) {
-                    String roleName = instanceProfile.getRoles().get(0).getArn();
-                    properties.put("roleArn", Strings.isNullOrEmpty(roleName) ? instanceProfile.getArn() : roleName);
-                }
-                cloudAccessConfigs.getCloudAccessConfigs().add(
-                        new CloudAccessConfig(
-                                instanceProfile.getInstanceProfileName(),
-                                instanceProfile.getInstanceProfileId(),
-                                properties));
+            String accessConfigType = filters.get(ACCESS_CONFIG_TYPE);
+            Set<CloudAccessConfig> cloudAccessConfigSet;
+            if ("ROLE".equals(accessConfigType)) {
```

Comment:
Can we use an Enum for this ?

Reply:
also can we add some logs here ?

Reply:
We should also add a log if the type is not ROLE or null then we will return all instance profiles